### ISSUE TITLE
[Bug]: Memory leak on thrift_server 

### ISSUE THREAD

**Issue by Ognimalf** [infiniflow/infinity#740]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

f3f9a24

### Actual behavior

After run python sdk test, LeakSanitizer report memory leak when shutdown service.
the LeakSanitize info:

```
Server is shutdown

=================================================================
==1012950==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 24 byte(s) in 1 object(s) allocated from:
    #0 0x5616fcbdfecd in operator new(unsigned long) (/home/xwg/dev/infinity/build/src/infinity+0x5faecd) (BuildId: a024c3f742ade692b48a462c2c7c7ea46c80cd5a)
    #1 0x5616fcf83b5a in infinity::InfinityServiceHandler@thrift_server::Insert(infinity_thrift_rpc::CommonResponse&, infinity_thrift_rpc::InsertRequest const&) /home/xwg/dev/infinity/src/network/thrift_server.cpp:209:24
    #2 0x5616fce10cd5 in infinity_thrift_rpc::InfinityServiceProcessor::process_Insert(int, apache::thrift::protocol::TProtocol*, apache::thrift::protocol::TProtocol*, void*) /home/xwg/dev/infinity/src/network/infinity_thrift/InfinityService.cpp:5990:13
    #3 0x5616fce08934 in infinity_thrift_rpc::InfinityServiceProcessor::dispatchCall(apache::thrift::protocol::TProtocol*, apache::thrift::protocol::TProtocol*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>> const&, int, void*) /home/xwg/dev/infinity/src/network/infinity_thrift/InfinityService.cpp:5639:3
    #4 0x5616fce523dd in apache::thrift::TDispatchProcessor::process(std::shared_ptr<apache::thrift::protocol::TProtocol>, std::shared_ptr<apache::thrift::protocol::TProtocol>, void*) /home/xwg/dev/infinity/third_party/thrift/lib/cpp/src/thrift/TDispatchProcessor.h:121:12
    #5 0x5616feaaa9d2 in apache::thrift::server::TConnectedClient::run() /home/xwg/dev/infinity/third_party/thrift/lib/cpp/src/thrift/server/TConnectedClient.cpp:61:24
    #6 0x5616fea6851b in apache::thrift::concurrency::ThreadManager::Task::run() /home/xwg/dev/infinity/third_party/thrift/lib/cpp/src/thrift/concurrency/ThreadManager.cpp:195:18
    #7 0x5616fea5f060 in apache::thrift::concurrency::ThreadManager::Worker::run() /home/xwg/dev/infinity/third_party/thrift/lib/cpp/src/thrift/concurrency/ThreadManager.cpp:310:19
    #8 0x5616fea9e259 in apache::thrift::concurrency::Thread::threadMain(std::shared_ptr<apache::thrift::concurrency::Thread>) /home/xwg/dev/infinity/third_party/thrift/lib/cpp/src/thrift/concurrency/Thread.cpp:28:23
    #9 0x5616fea9dc7d in void std::__invoke_impl<void, void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>(std::__invoke_other, void (*&&)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:61:14
    #10 0x5616fea9db1c in std::__invoke_result<void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>::type std::__invoke<void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>(void (*&&)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/invoke.h:96:14
    #11 0x5616fea9daf1 in void std::thread::_Invoker<std::tuple<void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>>::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:292:13
    #12 0x5616fea9dab4 in std::thread::_Invoker<std::tuple<void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>>::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:299:11
    #13 0x5616fea9d868 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(std::shared_ptr<apache::thrift::concurrency::Thread>), std::shared_ptr<apache::thrift::concurrency::Thread>>>>::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/std_thread.h:244:13
    #14 0x561700315982 in execute_native_thread_routine thread.o

SUMMARY: AddressSanitizer: 24 byte(s) leaked in 1 allocation(s).
```

### Steps to reproduce

```Markdown
run python tools/run_pysdk_test.py
```

**Comment by JinHai-CN** [comment]
Fixed by #744

**Comment by Ognimalf** [comment]
It looks fixed.
